StripeFinancialConnections/StripeFinancialConnections/Source/Common/TestModeAutofillBannerView.swift

```diff
@@ -0,0 +1,147 @@
+                )
+            case .otp:
+                STPLocalizedString(
+                    "Use test code",
+                    "Label for a button shown to a user who is in test mode, which will give them the option to autofill a mock one time passcode."
+                )
+            }
+        }
+    }
+
+    private let context: Context
+    private let didTapAutofill: () -> Void
+
+    // MARK: - Subviews
+
+    private lazy var messageLabel: UILabel = {
+        // Create icon as a text attachement.
+        let icon = Image.info
+            .makeImage(template: true)
+            .withTintColor(.attention300)
+        let textAttachment = NSTextAttachment(image: icon)
+
+        textAttachment.bounds = CGRect(
+            x: 0,
+            y: -2,
+            width: icon.size.width,
+            height: icon.size.height
+        )
+
+        let attributedString = NSMutableAttributedString()
+        attributedString.append(NSAttributedString(attachment: textAttachment))
+        attributedString.append(NSAttributedString(string: STPLocalizedString(
+            " You're in test mode.",
+            "Message shown to a user who is in test mode, which will give them the option to autofill mock credentials. There is intentionally a leading space here to provide a bit of spacing between the text and the icon."
+        )))
+
+        let label = UILabel()
+        label.attributedText = attributedString
+        label.font = .preferredFont(forTextStyle: .body)
+        label.textColor = .textDefault
+        return label
+    }()
+
+    private lazy var autofillDataButton: UIButton = {
+        let button = UIButton()
+        let title = context.buttonLabel
+        button.setTitle(title, for: .normal)
+        button.setTitleColor(.textActionPrimary, for: .normal)
+        button.titleLabel?.textAlignment = .right
+        button.titleLabel?.font = .preferredFont(forTextStyle: .body, weight: .semibold)
+        button.addTarget(self, action: #selector(autofillTapped), for: .touchUpInside)
+        return button
+    }()
+
+    private let stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.axis = .horizontal
+        stackView.distribution = .fill
+        return stackView
+    }()
+
+    // MARK: - Init
+
+    init(context: Context, didTapAutofill: @escaping () -> Void) {
+        self.context = context
+        self.didTapAutofill = didTapAutofill
+        super.init(frame: .zero)
+
+        backgroundColor = .attention50
+        layer.cornerRadius = 12
+        clipsToBounds = true
+
+        autofillDataButton.setContentCompressionResistancePriority(.required, for: .horizontal)
+
+        stackView.addArrangedSubview(messageLabel)
+        stackView.addArrangedSubview(autofillDataButton)
+
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(stackView)
+
+        NSLayoutConstraint.activate([
+            heightAnchor.constraint(equalToConstant: 38),
```

Comment:
I think this might be an interesting test case on how this scales with dynamic font + potentially lengthy changes to text (or localization)

this is one reason why I often put everything in a UIStackView with padding because it automatically handles everything